apps/surfer/src/locations/ConfigScreen.tsx

```diff
@@ -0,0 +1,143 @@
+import { Heading, Box, Paragraph, List, Note, TextLink } from '@contentful/f36-components';
+import tokens from '@contentful/f36-tokens';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import { ContentTypeProps } from 'contentful-management';
+import { useEffect, useState } from 'react';
+import { css } from '@emotion/css';
+import { FieldSelection } from '../components/FieldSelection';
+import { useSurferCompatibility } from '../hooks/useSurferCompatibility';
+import { useFieldSelection } from '../hooks/useFieldSelection';
+import { SurferLogotype } from '../components/SurferLogotype';
+
+const styles = {
+  splitter: css({
+    marginTop: tokens.spacingL,
+    marginBottom: tokens.spacingL,
+    border: 0,
+    height: '2px',
+    backgroundColor: tokens.gray300,
+  }),
+  body: css({
+    height: 'auto',
+    minHeight: '65vh',
+    margin: '0 auto',
+    marginTop: tokens.spacingXl,
+    padding: `${tokens.spacingXl} ${tokens.spacing2Xl}`,
+    maxWidth: '900px',
+    backgroundColor: tokens.colorWhite,
+    zIndex: 2,
+    boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.1)',
+    borderRadius: '2px',
+  }),
+  background: css({
+    display: 'block',
+    position: 'absolute',
+    zIndex: -1,
+    top: 0,
+    width: '100%',
+    height: '300px',
+    backgroundColor: '#250956',
+  }),
+};
+
+const Splitter = () => {
+  return <hr className={styles.splitter} />;
+};
+
+const embedInSidebar = (selectedContentTypes: string[]) => {
+  return {
+    EditorInterface: selectedContentTypes.reduce((acc: AppState['EditorInterface'], id) => {
+      acc[id] = { sidebar: { position: 1 } };
+
+      return acc;
+    }, {}),
+  };
+};
+
+export const ConfigScreen = () => {
+  const [contentTypes, setContentTypes] = useState<ContentTypeProps[]>();
+
+  const compatibility = useSurferCompatibility(contentTypes);
+  const { selectMany, selectAllCompatible, selectedContentTypes, toggleContentType, toggleField, selectedContentFields } = useFieldSelection(compatibility);
+  const sdk = useSDK<ConfigAppSDK>();
+
+  useEffect(() => {
+    sdk.cma.contentType.getMany({ spaceId: sdk.ids.space }).then((contentTypes) => {
+      setContentTypes(contentTypes.items);
+    });
+  }, [sdk.app, sdk.cma.contentType, sdk.ids.space]);
+
+  useEffect(() => {
+    if (contentTypes && compatibility) {
+      sdk.app.getCurrentState().then((appState) => {
+        if (appState?.EditorInterface) {
+          selectMany(Object.keys(appState.EditorInterface), sdk.parameters.installation.selectedContentFields ?? compatibility.compatibleFields);
+        } else {
+          selectAllCompatible();
+        }
+
+        sdk.app.setReady();
+      });
+    }
+  }, [contentTypes, compatibility, sdk.app, selectMany, selectAllCompatible, sdk.parameters.installation.selectedContentFields]);
+
+  sdk.app.onConfigure(() => {
+    return {
+      targetState: embedInSidebar(selectedContentTypes),
+      parameters: {
+        selectedContentFields,
+      },
+    };
+  });
```

Comment:
Shouldn't this be inside of `useEffect`? 🤔

Reply:
Ha! Doesn't have to be (it's registering a callback, not doing the configuration), but it's a good idea and it's misleading in the code as well. Fixed